qiskit/opflow/gradients/natural_gradient.py

```diff
@@ -120,50 +113,76 @@ def convert(
         # Instantiate the QFI metric which is used to re-scale the gradient
         metric = self._qfi_method.convert(operator[-1], params) * 0.25
 
-        # Define the function which compute the natural gradient from the gradient and the QFI.
         def combo_fn(x):
-            c = np.real(x[0])
-            a = np.real(x[1])
-            if self.regularization:
-                # If a regularization method is chosen then use a regularized solver to
-                # construct the natural gradient.
-                nat_grad = NaturalGradient._regularized_sle_solver(
-                    a, c, regularization=self.regularization
-                )
-            else:
-                try:
-                    # Try to solve the system of linear equations Ax = C.
-                    nat_grad = np.linalg.solve(a, c)
-                except np.linalg.LinAlgError:  # singular matrix
-                    nat_grad = np.linalg.lstsq(a, c)[0]
-            return np.real(nat_grad)
+            return self.nat_grad_combo_fn(x, self.regularization)
 
         # Define the ListOp which combines the gradient and the QFI according to the combination
         # function defined above.
         return ListOp([grad, metric], combo_fn=combo_fn)
 
+    @staticmethod
+    def nat_grad_combo_fn(x, regularization=None):
+        """
+        Natural Gradient Function Implementation.
+        Args:
+            x: Iterable consisting of Gradient, Quantum Geometric Tensor.
+            regularization: Regularization method.
+
+        Returns: Natural Gradient.
+
+        """
+        gradient = x[0]
+        metric = x[1]
+        if any(np.abs(np.imag(c_item)) > 1e-8 for c_item in gradient):
+            raise ValueError("The imaginary part of the gradient are non-negligible.")
+        if np.any([[np.abs(np.imag(a_item)) > 1e-8 for a_item in a_row] for a_row in metric]):
+            raise ValueError("The imaginary part of the metric are non-negligible.")
```

Comment:
this cannot happen anymore now that we have the check in the QFI right?

Reply:
Right. Good point.